test/Microsoft.Extensions.DiagnosticAdapter.Test/DiagnosticSourceAdapterTest.cs

```diff
@@ -70,6 +73,26 @@ public void IsEnabled_True_PredicateCalledForIsEnabled()
         }
 
         [Fact]
+        public void IsEnabled_True_PredicateCalledForIsEnabled_WithContext()
+        {
+            // Arrange
+            var callCount = 0;
+            Func<string, object, object, bool> isEnabled = (name, arg1, arg2) =>
+            {
+                Assert.Equal("One", name);
+                Assert.Equal("Target info", arg1);
+                callCount++;
+                return true;
+            };
```

Comment:
Add a test where you register a `Func<string, bool>` but call `IsEnabled(string, object, object)